tasks/vagrant.py

```diff
@@ -26,6 +26,100 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def with_vagrant_ad(func):
+    def wrapper(self, *args, **kwargs):
+        try:
+            __setup_provision_ad(self)
+        except TaskException as exc:
+            logging.critical('vagrant or provisioning failed')
+            raise exc
+        else:
+            func(self, *args, **kwargs)
+        finally:
+            if not self.no_destroy:
+                self.execute_subtask(
+                    VagrantCleanup(raise_on_err=False))
+
+    return wrapper
+
+
+def retry(times=1):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            counter = 0
+            while counter < times:
+                counter += 1
+                try:
+                    func(*args, **kwargs)
+                except:
+                    pass
+
+        return wrapper
+    return decorator
+
+
+def __setup_provision_ad(task):
+    """
+    This tries to execute the provision twice due to
+    problems described in issue #20
+    """
+    task.execute_subtask(
+        VagrantBoxDownload(
+            box_name=task.template_name,
+            box_version=task.template_version,
+            link_image=task.link_image,
+            timeout=None))
+
+    def setup_ad_machines():
+        task.execute_subtask(VagrantUpADRoot(timeout=None))
+        task.execute_subtask(VagrantUpADChild(timeout=None))
+
+    def prepare_ansible_config():
+        cfg_path = '.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory'
+        task.execute_subtask(PopenTask(['mkdir', '-p', '.vagrant/provisioners/ansible/inventory/']))
+        task.execute_subtask(PopenTask(['touch', cfg_path]))
+
+    def cfg_for_windows():
+        cfg_path = '.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory'
```

Comment:
I don't have this file. Is it created by running vagrant manually? Is it just the IP address of the `ad` host?